### DISCUSSION TITLE
What mathematical primitives could be further supported

### DISCUSSION THREAD

**Discussion by chrishobcroft** [PaulRBerg/prb-math#162] [Ideas]
This is quite a general question, cognicent that there are libraries in Solidity emerging for things like complex arithmetic amongst other things.

The topic is: what other mathematical primitives could be further supported in Solidity.

One idea: geometric arithmetic, e.g. scalar / vector products.

**Comment by PaulRBerg** [comment]
Thanks for starting this discussion, @chrishobcroft. General topics are welcome!

Here are the mathematical primitives that I think would be useful for DeFi:

- [ ] Linear Algebra (what you suggested)
- [ ] Complex numbers
- [ ] Trigonometry (Matt Solomon has a library for this: [solidity-trigonometry](https://github.com/mds1/solidity-trigonometry))

FWIW, I am toying with the idea of adding support for complex numbers in PRBMath by adding two new value types, `C59x18` and `U60x18`.
